xdsl/dialects/llvm.py

```diff
@@ -633,6 +634,13 @@ class ICmpPredicateAttr(SpacedOpaqueSyntaxAttribute, EnumAttribute[ICmpPredicate
     def from_int(cls, i: int):
         return ICmpPredicateAttr(cls.ALL_PREDICATES[i])
 
+    @classmethod
+    def int_from(cls, flag: ICmpPredicateFlag) -> int:
+        return cls.ALL_PREDICATES.index(flag)
```

Comment:
Can you call this function `to_int` and add it as a method of `ICmpPredicateFlag` directly?

Reply:
I would rather not, since I'm using `ALL_PREDICATES`

Reply:
What is the point of the attribute if it's never used in the IR? Could the helpers not be on the enum directly?

Reply:
Also, this is also O(n), I would recommend doing something like this:
```suggestion
    INDEX_BY_PREDICATE = {f: i for (i, f) in enumerate(ALL_PREDICATES)}

    @classmethod
    def int_from(cls, flag: ICmpPredicateFlag) -> int:
        return INDEX_BY_PREDICATE[flag]
```